pyodide-build/pyodide_build/mkpkg.py

```diff
@@ -119,11 +166,106 @@ def run_prettier(meta_path):
     subprocess.run(["npx", "prettier", "-w", meta_path])
 
 
+def _make_project_compare_name(project_name):
+    project_name_cleaned = project_name.replace("_", "").replace("-", "").lower()
+    if project_name_cleaned.startswith("py"):
+        project_name_cleaned = project_name[2:]
+    return project_name_cleaned
+
+
+# download a package from pypi and guess what modules are provided by it
+def _find_imports_from_package(url, project_name):
+    project_name_cleaned = _make_project_compare_name(project_name)
+    all_imports = []
+    try:
+        if url.endswith(".zip") or url.endswith(".whl"):
+            filetype = ".zip"
+        elif url.endswith(".tar.gz") or url.endswith(".tgz"):
+            filetype = ".tar.gz"
+        else:
+            print("Unknown archive type for {url}, can't determine imports")
+            return []
+        with tempfile.NamedTemporaryFile(suffix=filetype) as tf:
+            with urllib.request.urlopen(url) as fd:
+                tf.write(fd.read())
+            all_files = []
+            if url.endswith(".zip") or url.endswith(".whl"):
+                # wheel is zip
+                with ZipFile(tf) as package:
+                    all_files = package.namelist()
+            elif url.endswith(".tar.gz") or url.endswith(".tgz"):
+                # wheel is tar gz
+                with tarfile.open(tf.name) as package:
+                    all_files = package.getnames()
+                    all_modules = []
+            all_modules = []
+            for f in all_files:
+                # find module folders
+                if f.endswith("__init__.py") and f.find("/test/") == -1:
+                    all_modules.append(PurePath("/" + f).parents[0])
+            if len(all_modules) == 0:
+                # is this maybe a single file module with a file called 'modulename.py'?
+                for f in all_files:
+                    p = PurePath("/" + f)
+                    if p.suffix == ".py":
+                        cleaned = _make_project_compare_name(p.stem)
+                        if cleaned == project_name:
+                            return [p.stem]
+            if len(all_modules) == 0:
+                print(f"WARNING: COuldn't find any imports in package {url}")
+                return []
+            main_prefix = None
+            main_prefix_level = None
+            # check for first bit of the path that looks like the package name
+            # without any _ or -
+            for mod in all_modules:
+                level = len(mod.parts)
+                path_part = mod.parts[-1]
+                cleaned = _make_project_compare_name(path_part)
+                if cleaned == project_name_cleaned:
```

Comment:
I dont think this would be very reliable e.g.

 - scikit-learn that import as pandas
 - pyarrow that import as arrow
 - etc.

Maybe just listing all the modules with an `__init__.py` (e.g. found by `find_packages`) and letting the user remove the extra ones would be more reliable.